### PR TITLE
feat [#127] : support compression gzip, deflate

### PR DESCRIPTION
**Pull request by ariskemper**
it fix issue #127 with using compress middleware on node.js https://github.com/honojs/hono/issues/2061

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (yusukebe) [inline .vscode/launch.json:1]: Is this file necessary?

**Comment by yusukebe** [comment]
Hi @ariskemper !

Thanks so much. Looks great to me!

@usualoma Could you review this?

**Comment by usualoma** [comment]
Hi @ariskemper 
Thanks for creating the PR!

But I think the problem with https://github.com/honojs/hono/issues/2061 is not with "node-server" but with "core". It should be fixed once the following PR is merged.
https://github.com/honojs/hono/pull/2080

cc: @yusukebe
